apps/remote_control/lib/lexical/remote_control/build/document.ex

```diff
@@ -1,11 +1,36 @@
 defmodule Lexical.RemoteControl.Build.Document do
   alias Lexical.Document
+  alias Lexical.RemoteControl.Build
   alias Lexical.RemoteControl.Build.Document.Compilers
 
   @compilers [Compilers.Config, Compilers.Elixir, Compilers.EEx, Compilers.HEEx, Compilers.NoOp]
 
   def compile(%Document{} = document) do
     compiler = Enum.find(@compilers, & &1.recognizes?(document))
-    compiler.compile(document)
+    me = self()
+
+    {pid, ref} =
+      spawn_monitor(fn ->
+        result = compiler.compile(document)
+        send(me, {:result, result})
+      end)
+
+    receive do
+      {:result, result} ->
+        flush_normal_down(ref, pid)
+        result
+
+      {:DOWN, ^ref, :process, ^pid, {exception, stack}} ->
```

Comment:
Are we sure the reason will always be of the shape `{exception, stack}`?

Reply:
For the condition that I'm trying to solve, yes.

Reply:
...but I can't be sure that in the future we won't have some other shape. I was hoping to let it crash and have people report a bug if it does, rather than silently swallowing errors.

Reply:
It's not going to crash unless we crash it though, right? It'll just receive forever?

Reply:
ah, good point, I'll make it fail a pattern match

Reply:
Now that I introduced the Isolation module, it'll crash in the callers with a case clause error.